src/ts/common/ui/modal.ts

```diff
@@ -97,12 +96,12 @@ const addOnClick = (element: HTMLElement, exit: () => void, onClick?: () => Prom
 	element.addEventListener("click", () => callback().finally(exit));
 };
 
-const createModal = ({text, subText, elements, onCancel, colour, exitable = true}: ModalOptions): void => {
```

Comment:
`exitable` can just be implemented via recursion, just like you did with `onClick`